src/Components/CreateImageWizard/validators.ts

```diff
@@ -67,6 +67,21 @@ export const isFileSystemConfigValid = (partitions: Partition[]) => {
   return duplicates.length === 0;
 };
 
+export const isUserNameValid = (userName: string) => {
+  if (userName === undefined) return false;
+  const isLengthValid = userName.length >= 1 && userName.length <= 32;
+
+  const isNotNumericOnly = !/^\d+$/.test(userName);
+  const isPatternValid = /^[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9_$]$/.test(
+    userName
+  );
+  const doesNotStartWithDash = !/^[-]/.test(userName);
```

Comment:
This is already included in `isPatternValid`. The first character is limited to `[a-zA-Z0-9]`

Reply:
The `doesNotStartWithDash` I mean

Reply:
Also the min length based on `isPatternValid` is already two characters. The `userName.length >= 1` in `isLengthValid` is not needed.